src/logger/src/logger.rs

```diff
@@ -420,7 +420,7 @@ impl Logger {
                 // No need to explicitly call flush because the underlying LineWriter flushes
                 // automatically whenever a newline is detected (and we always end with a
                 // newline the current write).
-                if guard.write(&(format!("{}\n", msg)).as_bytes()).is_err() {
+                if guard.write_all(&(format!("{}\n", msg)).as_bytes()).is_err() {
```

Comment:
Why switch to `write_all`? My understanding is that depending on the writer it can be more inefficient than `write` + `flush`.

Reply:
From what I understand, the intention here is to write an entire log record. This guarantee can only be satisfied by `write_all`. `write` can write only a part of the log and return with `Ok(num_written_bytes)`. This can lead to discarding part of the log record. `flush` shouldn't matter here. `write_all` doesn't necessarily lead to a flush. `flush` doesn't necessarily have to be called after `write`.

This may affect performance or not. Depends on the implementation of the writer. But I think that we should offer the guarantee that the entire log message is written.

Reply:
I think we are calling flush after each write anyway. But I guess it makes sense to call write_all instead. Can you also specify that you are using write_all instead of write in the commit message as well?

Reply:
Sure. Done.